File: PennyTracker.Frontend/src/models/ExpenseInsight.ts
import { TransactionCategory } from "../enums/TransactionCategory";
import { Transaction } from "./Transaction";

export interface ExpenseInsight {
    totalExpenses: number;
    totalIncomes: number;
    transactions: Transaction[];
    categoryWiseExpenses: Record<TransactionCategory, number>;
    categoryWiseIncomes: Record<TransactionCategory, number>;
    monthlyExpenses: Record<string, number>;
    monthlyIncomes: Record<string, number>;
}

File: PennyTracker.Backend/Models/ExpenseInsight.cs
using PennyTracker.Backend.Enums;

namespace PennyTracker.Backend.Models
{
    public class ExpenseInsight
    {
        public List<Transaction> Transactions { get; set; } = [];

        public decimal TotalExpenses { get; set; }

        public decimal TotalIncomes { get; set; }

        public Dictionary<TransactionCategory, decimal> CategoryWiseExpenses { get; set; } = [];

        public Dictionary<TransactionCategory, decimal> CategoryWiseIncomes { get; set; } = [];

        public Dictionary<Month, decimal> MonthlyExpenses { get; set; } = [];

        public Dictionary<Month, decimal> MonthlyIncomes { get; set; } = [];
    }
}
